weatherAplication/ShowWeather/ShowWeatherViewController.swift

```diff
@@ -15,65 +13,31 @@ class ShowWeatherViewController: UIViewController {
     private let cityLabel = UILabel()
     private let weatherTableView = UITableView()
 
-    private var date = String()
+    private var dateString = String()
 
     private var dataArray: [WeatherData] = []
-    private let urlString: String = "https://api.open-meteo.com/v1/forecast?"
-
-    func makeURLRequest() {
-        guard let queryURL = URL(string: urlString + "latitude=\(latitude)&longitude=\(longitude)&hourly=temperature_2m,surface_pressure,weathercode,windspeed_10m") else { return }
-        let session = URLSession.shared
-
-        session.dataTask(with: queryURL, completionHandler: { [weak self] data, response, error -> Void in
-
-            if error != nil {
-                let alert = UIAlertController(title: "Error", message: error?.localizedDescription, preferredStyle: .alert)
-                alert.addAction(UIAlertAction(title: "OK", style: .default))
-                self?.present(alert, animated: true)
-            }
-
-            if let httpResponse = response as? HTTPURLResponse {
-                if 200...299 ~= httpResponse.statusCode {
-                    do {
-                        guard let data = data else { return }
-                        let jsonResult = try JSONDecoder().decode(OpenMeteoDecoded.self, from: data)
-                        self?.takeDataFromJson(jsonResult: jsonResult)
-                        DispatchQueue.main.async {
-                            self?.weatherTableView.reloadData()
-                        }
-                    } catch let error {
-                        print(error)
-                    }
-                }
-            }
-
-        }).resume()
-    }
-
-    func takeDataFromJson(jsonResult: OpenMeteoDecoded) {
-        var dataArray: [WeatherData] = []
-        if jsonResult.hourly.time.count >= kElementsToShow,
-           jsonResult.hourly.temperature.count >= kElementsToShow,
-           jsonResult.hourly.surfacePressure.count >= kElementsToShow,
-           jsonResult.hourly.windspeed.count >= kElementsToShow,
-           jsonResult.hourly.weathercode.count >= kElementsToShow {
-            date = String(jsonResult.hourly.time[0].prefix(kCharsToDrop - 1))
-            for index in 0...(kElementsToShow - 1) {
-                dataArray.append(WeatherData(hour: String(jsonResult.hourly.time[index].dropFirst(kCharsToDrop)),
-                                             temperature: jsonResult.hourly.temperature[index],
-                                             pressure: jsonResult.hourly.surfacePressure[index],
-                                             windSpeed: jsonResult.hourly.windspeed[index],
-                                             weatherCode: jsonResult.hourly.weathercode[index]))
-            }
-            self.dataArray = dataArray
-        }
-    }
+    private let apiClient = APIClient.sharedInstance
 
     override func viewDidLoad() {
```

Comment:
może to drobna rzecz ale funkcję viewDidLoad możesz przenieść poniżej inita. W taki sposób trochę łatwiej się to czyta bo funkcje są poukładane w kolejności ich wykonywania